compiler/src/typed/module_resolution.re

```diff
@@ -359,31 +361,38 @@ module Dependency_graph =
           // all dependencies have expected CRC, and the module was compiled with
           // the current compiler configuration. Otherwise, we need to recompile.
           let config_sum = Cmi_format.config_sum();
-          let cmi = read_file_cmi(objpath);
           let base_dir = Filename.dirname(srcpath);
           dn.dn_up_to_date :=
-            config_sum == cmi.cmi_config_sum
-            && file_older(srcpath, objpath)
-            && List.for_all(
-                 ((name, crc)) => {
-                   let resolved = resolve_unit(~base_dir, name);
-                   let out_file_name = get_output_name(resolved);
-                   Fs_access.file_exists(out_file_name)
-                   && (
-                     switch (crc) {
-                     | None => true
-                     | Some(crc) =>
-                       try(
-                         Cmi_format.cmi_to_crc(read_file_cmi(out_file_name))
-                         == crc
-                       ) {
-                       | _ => true
-                       }
-                     }
-                   );
-                 },
-                 cmi.cmi_crcs,
-               );
+            (
+              switch (read_file_cmi(objpath)) {
+              | exception (Cmi_format.Error(_)) => false
+              | cmi =>
+                config_sum == cmi.cmi_config_sum
+                && file_older(srcpath, objpath)
+                && List.for_all(
+                     ((name, crc)) => {
+                       let resolved = resolve_unit(~base_dir, name);
+                       let out_file_name = get_output_name(resolved);
+                       Fs_access.file_exists(out_file_name)
+                       && (
+                         switch (crc) {
+                         | None => true
```

Comment:
Just a question, why isn't it considered not up to state if there's no crc?

Reply:
Good question. I think that's actually a quasi-bug since this is never actually `None`, but if it ever _were_ `None`, then this should be false. Since I fixed the other case I'll fix this one too.